### PR TITLE
[Fix] Flink connector support json import and use httpclient to streamlaod

### LINKED ISSUE #6550: doris-flink-connecter数据换行配置不起作用
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

1.0.0

### What's Wrong?

在使用doris-connector的时候，数据导入到doris一部分的试试报错：
image

最后发现问题如下：

mysql数据库中有个字段mateName是varchar（255），如下：
image
拷贝出来后内容，有换行：
image

然后，

image

但是 还是报一次样的错误。

### What You Expected?

希望设置换行符 sink.properties.line_delimiter 和 sink.properties.column_separator 能起作用。或者其他解决这个问题的办法。

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by JNSimba**
## Proposed changes

When the text is imported, there may be separators in the data;
Therefore, support json format import.
Need setting parameters (only json array format is provided, so it must be set in pairs)
**sink.properties.format = json
sink.properties.strip_outer_array=true**

And modify the HTTPURLConnection of streamload to Httpclient request

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #6550 ) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: extension/flink-doris-connector/pom.xml [hunk 1]
```diff
@@ -118,6 +118,21 @@
...
+                <exclusion>
+                    <artifactId>httpcore</artifactId>
+                    <groupId>org.apache.httpcomponents</groupId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>4.5.3</version>
```
Comment (hf200012) [inline extension/flink-doris-connector/pom.xml:135]: 4.5.13

**Review by hf200012 — COMMENTED** [review]
The http client version should be upgraded to 4.5.13

File: extension/flink-doris-connector/pom.xml (hunk 1 above)
Comment (JNSimba) [reply extension/flink-doris-connector/pom.xml]: modified

**Review by hf200012 — APPROVED** [review]
LGTM

**Review by hf200012 — DISMISSED** [review]

**Comment by mincwang** [comment]
Thanks @JNSimba for the nice contribution.

**Review by morningman — COMMENTED** [review]
Hi @JNSimba Maybe we can add this to the 
```
docs/zh-CN/extending-doris/flink-doris-connector.md
```
?

Add a simple instruction for this featuer?

**Comment by JNSimba** [comment]
> Hi @JNSimba Maybe we can add this to the
> 
> ```
> docs/zh-CN/extending-doris/flink-doris-connector.md
> ```
> 
> ?
> 
> Add a simple instruction for this featuer?

already add

**Review by hf200012 — APPROVED** [review]
LGTM

**Pull request merged**